index.js

```diff
@@ -131,6 +132,25 @@ function compression (options) {
       return this
     }
 
+    res.removeListener = function removeListener (type, listener) {
```

Comment:
So, I think that this may open a new bug, even though it fixes another bug: `res.addListener` is not overwritten. So during a compressed response, if `res.addListener('drain'...)` is used, this `removeListener` will now not removed that instance. This may actually be part of the `pipe` issue, as I think it uses `addListener` instead of `on`.